### PR TITLE
Flux conversions galore

### PR DESCRIPTION
**Pull request by SunilSimha**
The master branch needs some changes to make the flux conversions from mags less messy.

### CODE REVIEW

File: frb/surveys/catalog_utils.py [hunk 1]
```diff
@@ -264,6 +264,41 @@ def mag_from_flux(flux, flux_err=None):
...
+        mag_err (float, optional): uncertainty in magnitude
+    Returns:
+        flux (float): flux in mJy
+        flux_err (float): if mag_err is given, a corresponding
+            flux_err is returned.
+    """
+    # Data validation
+    assert np.isreal(mag), "Mags must be floats."
+    assert np.isreal(mag_err) | mag_err==None, "Mag errs must be floats"
+    try:
```
Comment (profxj) [inline frb/surveys/catalog_utils.py:284]: please avoid `try/except`

**Review by profxj — CHANGES_REQUESTED** [review]
just a minor nit

key question:  should we take the opportunity here to apply
this to all the FRB galaxies (hosts and foreground), i.e. 
update `build_hosts.py` and run it to re-create the JSON files?

**Comment by SunilSimha** [comment]
> just a minor nit
> 
> key question: should we take the opportunity here to apply
> this to all the FRB galaxies (hosts and foreground), i.e.
> update `build_hosts.py` and run it to re-create the JSON files?

I think it would be nice to do that. We can have separate columns for the flux measurements in mags and in mJy. That way, the user will find the mags intuitive while also having no ambiguity with regards to the flux measurement. Alternatively, we can probably report the zero-point mags for each band.

**Comment by profxj** [comment]
yeah, I like separate columns, even if it means 2x more
data in the file.  Do you want to take that on, or should
I do it?

**Comment by profxj** [comment]
ok @SunilSimha -- I re-built the hosts.

Can you spot check a few before I merge?
Thanks

**Comment by SunilSimha** [comment]
Fluxes look alright and the code seems to handle non-detections as expected. I think this is good to merge.

**Review by profxj — APPROVED** [review]
Time to merge!

**Pull request merged**